src/applications/personalization/profile-2/tests/e2e/profile.cypress.spec.js

```diff
@@ -0,0 +1,105 @@
+import { PROFILE_PATHS } from '../../constants';
+
+import mockUser from '../fixtures/users/user-36.json';
+import mockPaymentInfo from '../fixtures/payment-information/direct-deposit-is-set-up.json';
+import mockFeatureToggles from '../fixtures/feature-toggles.json';
+
+/**
+ *
+ * @param {boolean} mobile - test on a mobile viewport or not
+ *
+ * This helper:
+ * - loads the Profile, confirms the user is redirected to the correct URL, and
+ *   then performs an aXe scan
+ * - clicks through each item in the sub-nav, checks that the URL is correct,
+ *   and performs an aXe scan
+ */
+function checkAllPages(mobile = false) {
+  cy.visit('/profile');
+  if (mobile) {
+    cy.viewport('iphone-6');
+  }
+
+  // should show a loading indicator
+  cy.findByRole('progressbar').should('exist');
+  cy.findByText(/loading your information/i).should('exist');
+
+  // and then the loading indicator should be removed
+  cy.findByRole('progressbar').should('not.exist');
+  cy.findByText(/loading your information/i).should('not.exist');
+
+  // visiting /profile should redirect to profile/personal-information
+  cy.url().should(
+    'eq',
+    `${Cypress.config().baseUrl}${PROFILE_PATHS.PERSONAL_INFORMATION}`,
+  );
+
+  // make the a11y check on the Personal Info section
+  cy.injectAxe();
+  cy.checkA11y();
+
+  // make the  a11y check on the Military Info section
+  if (mobile) {
```

Comment:
What's your opinion on splitting up the test files and separate out mobile to avoid a bunch of conditional logic?

Reply:
Well this was my attempt at reducing a bunch of code that was basically the same. So it's either have this conditional logic (or something similar) or have a bunch of repetitive code.